pkg/agent/flowexporter/exporter/exporter_test.go

```diff
@@ -53,32 +54,39 @@ func TestFlowExporter_sendTemplateRecord(t *testing.T) {
 	// Only the element name is needed, other arguments have dummy values.
 	elemList := make([]*ipfixentities.InfoElement, 0)
 	for _, ie := range IANAInfoElements {
-		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, 0, 0))
+		element, err := ipfixregistry.GetInfoElement(ie, ipfixregistry.IANAEnterpriseID)
+		if err != nil {
+			t.Errorf("Information Element %s in IANA registry does not exist.", ie)
+		}
+		elemList = append(elemList, element)
 	}
 	for _, ie := range IANAReverseInfoElements {
-		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, antreaEnterpriseRegistry, 0))
+		element, err := ipfixregistry.GetInfoElement(ie, ipfixregistry.ReverseEnterpriseID)
+		if err != nil {
+			t.Errorf("Reversed Information Element %s in IANA registry does not exist.", ie)
+		}
+		elemList = append(elemList, element)
 	}
 	for _, ie := range AntreaInfoElements {
-		elemList = append(elemList, ipfixentities.NewInfoElement(ie, 0, 0, 0, 0))
+		element, err := ipfixregistry.GetInfoElement(ie, ipfixregistry.AntreaEnterpriseID)
+		if err != nil {
+			t.Errorf("Information Element %s in Antrea registry does not exist.", ie)
+		}
+		elemList = append(elemList, element)
 	}
 	// Expect calls for different mock objects
 	tempBytes := uint16(0)
 	var templateRecord ipfixentities.Record
 
 	mockTempRec.EXPECT().PrepareRecord().Return(tempBytes, nil)
-	for i, ie := range IANAInfoElements {
-		mockIPFIXExpProc.EXPECT().GetIANARegistryInfoElement(ie, false).Return(elemList[i], nil)
+	for i, _ := range IANAInfoElements {
+		t.Log(tempBytes)
 		mockTempRec.EXPECT().AddInfoElement(elemList[i], nil).Return(tempBytes, nil)
 	}
-	for i, ie := range IANAReverseInfoElements {
-		split := strings.Split(ie, "_")
-		runeStr := []rune(split[1])
-		runeStr[0] = unicode.ToLower(runeStr[0])
-		mockIPFIXExpProc.EXPECT().GetIANARegistryInfoElement(string(runeStr), true).Return(elemList[i+len(IANAInfoElements)], nil)
+	for i, _ := range IANAReverseInfoElements {
 		mockTempRec.EXPECT().AddInfoElement(elemList[i+len(IANAInfoElements)], nil).Return(tempBytes, nil)
 	}
-	for i, ie := range AntreaInfoElements {
-		mockIPFIXExpProc.EXPECT().GetAntreaRegistryInfoElement(ie, false).Return(elemList[i+len(IANAInfoElements)+len(IANAReverseInfoElements)], nil)
```

Comment:
Can't we add the GetInfoElement method from the IPFIX registry mock here? Idea is to use less third-party library code in unit tests.